hack/generator/pkg/astmodel/struct_type.go

```diff
@@ -23,18 +23,26 @@ var EmptyStructType = NewStructType()
 var _ Type = (*StructType)(nil)
 
 // NewStructType is a factory method for creating a new StructTypeDefinition
-func NewStructType(fields ...*FieldDefinition) *StructType {
-	sort.Slice(fields, func(left int, right int) bool {
-		return fields[left].fieldName < fields[right].fieldName
-	})
-
-	return &StructType{fields, make(map[string]Function)}
+func NewStructType() *StructType {
+	return &StructType{
+		fields:    make(map[FieldName]*FieldDefinition),
+		functions: make(map[string]Function),
+	}
 }
 
 // Fields returns all our field definitions
-// A copy of the slice is returned to preserve immutability
+// A sorted copy of the slice is returned to preserve immutability and provide determinism
```

Comment:
minor: Since there is no longer "a slice" by default, might be worth rephrasing this to not say: "a sorted copy of the slice"
```suggestion
// A sorted slice is returned to preserve immutability and provide determinism
```